snippets/puppet_setup.erb

```diff
@@ -0,0 +1,48 @@
+<%#
+kind: snippet
+name: puppet_setup
+description: this snippet will configure the Puppet agent
+%>
+<%
+etc_path = (@host.operatingsystem.family == 'Freebsd') ? '/usr/local/etc/puppet' : '/etc/puppet'
+bin_path = (@host.operatingsystem.family == 'Freebsd') ? '/usr/local/bin' : '/usr/bin'
+%>
+
+<% if @host.operatingsystem.family == 'Debian' -%>
+apt-get update
+apt-get install -y puppet
+<% elsif @host.operatingsystem.family == 'Freebsd' -%>
+pkg install -y puppet
+<% elsif @host.operatingsystem.family == 'Redhat' -%>
+yum -t -y -e 0 install puppet
+<% end -%>
+
+cat > <%= etc_path %>/puppet.conf << EOF
+<%= snippet 'puppet.conf' %>
+EOF
+
+<% if @host.operatingsystem.family == 'Debian' -%>
+if [ -f "/etc/default/puppet" ]
+then
+/bin/sed -i 's/^START=no/START=yes/' /etc/default/puppet
+fi
+/usr/bin/puppet agent --enable
+<% elsif @host.operatingsystem.family == 'Freebsd' -%>
+echo 'puppet_enable="YES"' >>/etc/rc.conf
+<% elsif @host.operatingsystem.family == 'Redhat' -%>
```

Comment:
You could perhaps put the `elsif os.name == 'Fedora'` first, then have this `elsif os.family == 'RedHat'` below to avoid an extra level of conditionals.